app/src/main/java/com/example/composesigninsample/MainActivity.kt

```diff
@@ -49,11 +60,35 @@ fun AppRouter() {
     val navController = rememberNavController()
     NavHost(navController = navController, startDestination = Screen.HomeScreen.route) {
         composable(Screen.HomeScreen.route) {
-            HomeScreen({
-                navController.navigate(Screen.SignInScreen.route)
-            })
+            val homeViewModel: HomeViewModel = hiltViewModel()
+
+            HomeScreen(
+                navigateToSignIn = {
+                    navController.navigate(Screen.SignInScreen.route) {
+                        popUpTo(Screen.HomeScreen.route) {
+                            inclusive = true
+                        }
+                    }
+                },
+                signOut = homeViewModel::signOut,
+                isSignedIn = homeViewModel.loggedInState,
+            )
+        }
+        composable(Screen.SignInScreen.route) {
+            val signInViewModel: SignInViewModel = hiltViewModel()
+
+            SignInScreen(
+                navigateToHome = {
+                    navController.navigate(Screen.HomeScreen.route) {
+                        popUpTo(Screen.SignInScreen.route) {
+                            inclusive = true
+                        }
+                    }
+                },
+                setLoggedIn = signInViewModel::signIn,
+                isSignedIn = signInViewModel.loggedInState,
```

Comment:
I like this approach so that SignInScreen has no dependency on SignInViewModel

Reply:
Yep, same